style-guide: document use of POSIX-compliant shell syntax (#12894)
tldr-pages/tldr: contributing-guides/style-guide.md

Author: Editing mistake

## Example commands

> - The examples of pages in the subdirectory `common` for UNIX-like operating systems (e.g. `pages/common/tar.md`), preferably, should be written with the shell syntax specified by the [latest POSIX standard](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18).

Reviewer: ```suggestion
### POSIX-compliant syntax

- The examples of pages in the subdirectory `common` for UNIX-like operating systems (e.g. `pages/common/tar.md`), preferably, should be written with the shell syntax specified by the [latest POSIX standard](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18).
```

We generally add all contents behind a third-level heading in the style guide since the 2nd level heading is used for the main topics.

----

Also, I suggest moving this to the last part of this section to not confuse new users (who don't know about the POSIX standard).

  However, if you believe the example is hard to read, has little to no use, or is impossible to write in POSIX-compliant syntax, use non-POSIX syntax and add a note or warning indicating it and some or all of its consequences concisely and specifically. For example:

  ```md
  # IFS

  > IFS (Internal Field Separator) is a special environment variable that defines the delimiter used for word splitting in Unix shells.
  > The default value of IFS is a space, tab, and newline. The three characters serve as delimiters.

>   > Note: Not all shells suport ANSI C quoted strings (i.e., $'...' and $"..."), the dollar sign could be included in the string.

Reviewer: ```suggestion
  > Note: Not all shells support ANSI C quoted strings (i.e., $'...' and $"..."), the dollar sign could be included in the string.
```

  > More information: <https://www.gnu.org/software/bash/manual/html_node/Word-Splitting.html>.
  ```

>   To help you decide using POSIX-compliant syntax, ask yourself this:

Reviewer: Not sure about the wording of this section, maybe we can reword it a bit.

>   To help you decide using POSIX-compliant syntax, ask yourself this:
> 
>   - Is the example too long (i.e. [exceeds the maximum line length](https://github.com/tldr-pages/tldr/blob/main/.markdownlint.json#L5))?
>   - Does the example utilize any [risky](https://unix.stackexchange.com/a/278439) or [inefficient](https://unix.stackexchange.com/a/169765) commands?
>   - Does the example return the correct output?

Reviewer 2: ```suggestion
  As more of the points below apply, so does the recommendation to not use POSIX-compliant syntax in a particular example:

  - The example is not present in at least one implementation
  - The example is too long (i.e. [exceeds the maximum line length](https://github.com/tldr-pages/tldr/blob/main/.markdownlint.json#L5))
  - The example utilize any [risky](https://unix.stackexchange.com/a/278439) or [inefficient](https://unix.stackexchange.com/a/169765) commands?
  - The example doesn't return the correct output in some or all implementations
```
How about this? @kbdharun @sebastiaanspeck @tricantivu

Reviewer 2: IMO questions makes things a bit too open and a bit confusing (a positive answer means that I should use POSIX-compliant syntax or not?)

Author: > IMO questions makes things a bit too open and a bit confusing (a positive answer means that I should use POSIX-compliant syntax or not?)

Perhaps I could replace the questions for a guide explaining how to check POSIX-compliance (e.g. `shellcheck -s sh -`).

  **Bad example**:

  *Fails with pathnames ending with trailing slashes*

  ```markdown
  - Get basename of UNIX pathname:

  `echo "${{{pathname}}##*/}"`
  ```

  **Good example**:

  ```md
  - Get basename of UNIX pathname:

  `basename "${{{pathname}}}"`

>   ```

Reviewer: Btw, what is this example trying to achieve? (maybe we can have a different one as this seems like our placeholder displaying syntax `/{/{`).

Author: My objective was to help readers distinguish when a command example written in POSIX-compliant syntax is bad or good in terms of usefulness and readability. Given the comment, I think my examples were not clear, I have removed them.
